### PR TITLE
feat: add footer property to Drawer

### LINKED ISSUE #15204: Drawer support footer
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
Currently Drawer not support footer which makes it hard to implement fixed Title & Footer but scrollable body.

### What does the proposed API look like?
```jsx
<Drawer className="fixed" title="2333" footer="66666">
 ...
</Drawer>
```

```less
.fixed {
  display: flex;
  ...

  .xxx-title, .xxx-footer {
    flex: none;
  }

  .xxx-body {
    flex: auto;
    overflow-y: auto;
  }
}
```

### PR DESCRIPTION
**Pull request by deanvanniekerk**
### 🤔 This is a ...

- [x] New feature
- [x] Bug fix
- [x] Site / document update

### 🔗 Related issue link

[feature request - Drawer footer support](https://github.com/ant-design/ant-design/issues/15204)

close #15204

### 💡 Background and solution

Currently in the [demo](https://ant.design/components/drawer/) the `Drawer` footer is broken. See below.

broken

I propose a solution to add a dedicated footer property on the `Drawer` component. This solution allows for a 'fixed' footer and title.

fixed

### 📝 Changelog

- Add footer property to Drawer

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Add footer property to Drawer   |
| 🇨🇳 Chinese |  将页脚属性添加到抽屉  |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/drawer/demo/form-in-drawer.md](https://github.com/DeanVanNiekerk/ant-design/blob/feature/components/drawer/demo/form-in-drawer.md)
[View rendered components/drawer/index.en-US.md](https://github.com/DeanVanNiekerk/ant-design/blob/feature/components/drawer/index.en-US.md)
[View rendered components/drawer/index.zh-CN.md](https://github.com/DeanVanNiekerk/ant-design/blob/feature/components/drawer/index.zh-CN.md)

### CODE REVIEW

File: components/drawer/style/drawer.less [hunk 1]
```diff
@@ -191,13 +191,27 @@
     background: @component-background;
   }
 
+  &-wrapper-body {
+    display: flex;
+    flex-direction: column;
+    flex-wrap: nowrap;
```
Comment (shaodahong) [inline components/drawer/style/drawer.less]: @jljsj33 This change merge branch maybe is 4.0?

Comment (jljsj33) [reply components/drawer/style/drawer.less]: 确定要加这玩意？？？现在的API口完全可以解决这个。。
https://stackblitz.com/edit/react-jjmuro?embed=1&file=index.js

Comment (jljsj33) [reply components/drawer/style/drawer.less]: @afc163

Comment (DeanVanNiekerk) [reply components/drawer/style/drawer.less]: > 确定要加这玩意？？？现在的 API 完全可以解决这个。。
> https://stackblitz.com/edit/react-jjmuro?embed=1&file=index.js

Setting `position: 'fixed', width: 720` for both **title** and **footer** doesn't seem like an elegant solution to me.

The width of 720 is a magic number that needs to match up with the width of the Drawer.

Comment (shaodahong) [reply components/drawer/style/drawer.less]: understand, that means easy to set a footer, like `Modal` footer, https://ant.design/components/modal/#API

Comment (jljsj33) [reply components/drawer/style/drawer.less]: 觉得有必要就加呗，不用发到 4.0，

width is not a magic number, required in drawers, default is 256.

Comment (jljsj33) [reply components/drawer/style/drawer.less]: 哦，还是发到 4.0 吧，，3.x 的实现方式就用我上面的吧，，4.x 改用这个，这个属于大变动了。。也好写个 chengelog

**Comment by zombieJ** [comment]
Using flex should add to `4.0-prepare` branch. Since v3 still need to support old version IE.

**Comment by DeanVanNiekerk** [comment]
@zombieJ should I create a new PR for this feature against the `4.0-prepare` branch? Then we can close this one?

**Comment by afc163** [comment]
Could you implement it without flex, then we can add this into 3.0.

**Comment by DeanVanNiekerk** [comment]
@afc163 I dont think there is an easy way to implement this without `flex`.

I will update this PR implement the fix to the demo as @jljsj33 suggested [here](https://stackblitz.com/edit/react-jjmuro?embed=1&file=index.js)

Then Ill created a new PR for `4.0-prepare` using `flex`

**Comment by afc163** [comment]
try 

```css
.ant-drawer-body {
  overflow: auto;
  height: calc(100vh - 100px); // 100px would be height of header + height of footer
}

**Comment by DeanVanNiekerk** [comment]
@afc163, the demo has already been fixed.
https://github.com/ant-design/ant-design/commit/0cb5601527f099b90c3846974f7c971d1db6ee20

I have created an new [PR](https://github.com/ant-design/ant-design/pull/19779) with the `flex` implementation for the `4.0-prepare` branch. 

This PR can now be closed.

**Comment by jljsj33** [comment]
> Try none flex css solution:
> 
> ```css
> .ant-drawer-body {
>   overflow: auto;
>   height: calc(100vh - 100px); // 100px would be height of header + height of footer
> }
> ```

这样不行的，，head 和 footer 不是固定有的，，要加 calc 也要在 js 里加，，而且需要给 footer 一个固定的样式和高度，

**Pull request closed without merging**